src/main/kotlin/com/terraformation/backend/accelerator/api/ProjectVotesController.kt

```diff
@@ -0,0 +1,129 @@
+  fun getProjectVotes(
+      @PathVariable("projectId") projectId: ProjectId,
+  ): GetProjectVotesResponsePayload {
+    return GetProjectVotesResponsePayload(projectId, "testProject", emptyList())
+  }
+
+  @ApiResponse200
+  @ApiResponse403
+  @ApiResponse404
+  @PutMapping
+  @Operation(
+      summary = "Upserts vote selections for a single project.",
+      description =
+          "Update the user's vote for the project phase. If the (user, project, phase) does not " +
+              "exist, a new entry is created. Setting a `voteOption` to `null` removes the vote.")
+  fun upsertProjectVotes(
+      @PathVariable("projectId") projectId: ProjectId,
+      @RequestBody payload: UpsertProjectVotesRequestPayload,
+  ): UpsertProjectVotesResponsePayload {
+    return UpsertProjectVotesResponsePayload(projectId, payload.phase, emptyList())
+  }
+
+  @ApiResponse200
+  @ApiResponse400
+  @ApiResponse403
+  @ApiResponse404
+  @DeleteMapping
+  @Operation(
+      summary = "Remove one or more voters from the project/phase.",
+      description =
+          "Remove the voters from the project phase, making them ineligible from voting. This is " +
+              "different from undoing a vote (by setting the `voteOption` to `null`). To remove " +
+              "voters from the entire project phase, set `userId` to `null`, and set " +
+              "`phaseDelete` to `true`")
+  fun deleteProjectVotes(
+      @PathVariable("projectId") projectId: ProjectId,
+      @RequestBody payload: DeleteProjectVotesRequestPayload,
+  ): SimpleSuccessResponsePayload {
+    return SimpleSuccessResponsePayload()
+  }
+}
+
+data class VoteSelection(
+    val userId: UserId,
+    @Schema(
+        description =
+            "The vote the user has selected. Can be yes/no/conditional or `null` if " +
+                "a vote is not yet selected.")
+    val voteOption: VoteOption? = null,
+    val conditionalInfo: String? = null,
+    val email: String? = null,
+    val firstName: String? = null,
+    val lastName: String? = null,
+)
+
+data class PhaseVotes(val phase: CohortPhase, val votes: List<VoteSelection>)
+
+data class UpsertVoteSelection(
+    val user: UserId,
+    @Schema(description = "If set to `null`, remove the vote the user has previously selected.")
+    val voteOption: VoteOption? = null,
+    val conditionalInfo: String? = null,
+)
+
+data class UpsertProjectVotesRequestPayload(
+    val projectId: ProjectId,
+    val phase: CohortPhase,
+    val votes: List<UpsertVoteSelection>
+)
+
+data class DeleteProjectVotesRequestPayload(
+    val projectId: ProjectId,
+    val phase: CohortPhase,
+    @Schema(description = "If set to `null`, all voters in the phase will be removed. ")
+    val userId: UserId? = null,
+    @Schema(
+        description =
+            "A safeguard flag that must be set to `true` for deleting all voters in " +
+                "a project phase. ")
+    val phaseDelete: Boolean = false,
```

Comment:
Do we set this to any specific value for a single user delete? If not, maybe make it optional.
I like this safeguard, is there a bulk delete use-case?

Reply:
I am thinking maybe in a case where a phase of vote is created incorrectly. There is not really a delete use case at all in the requirements.